File: 0011-container-with-most-water/0011-container-with-most-water.py
class Solution(object):
    def maxArea(self, height):
        area=0
        length=len(height)
        if length>2:
            i=0
            j=length-1
            while j>i:
                breadth=abs(i-j)
                a=min(height[i],height[j])
                b=breadth*a
                if b>area:
                    area=b
                if height[i]>height[j]:
                    j-=1
                else:
                    i+=1
        else:
            a=min(height[0],height[1])
            breadth=1
            b=breadth*a
            if b>area:
                area=b
        return area

File: 0011-container-with-most-water/0011-container-with-most-water.java
class Solution {
    public int maxArea(int[] height) {
        int n=height.length;
        int i=0;
        int j=n-1;
        int ans=0;
        int temp=0;
        if(n==2){
            return Math.min(height[0],height[1]);
        }
        while(i<j){
            temp=(Math.min(height[i],height[j]))*(j-i);
            ans=Math.max(ans,temp);
            if(height[i]>=height[j]){
                j--;
            }
            else{
                i++;
            }
        }
        return ans;
    }
}